cpp/src/parquet/arrow/reader.cc

```diff
@@ -781,6 +770,9 @@ Status FileReaderImpl::GetRecordBatchReader(const std::vector<int>& row_group_in
                                             const std::vector<int>& column_indices,
                                             std::unique_ptr<RecordBatchReader>* out) {
   // column indices check
+  ARROW_ASSIGN_OR_RAISE(std::ignore, manifest_.GetFieldIndices(column_indices));
```

Comment:
it would be nice if RETURN_NOT_OK could be used here.

Reply:
alright

Reply:
Just to clarify doesn't need to be for this pr